woocommerce/woocommerce-gutenberg-products-block
# Fix: Password Protection not respected on single product template

## What

Fixes #10939 

## Why

The function `add_password_form` adds the password form to the single product template when the first block that belongs to the single product block is founded. The rest of the blocks related to the Single Product template are removed.





## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Ensure you are using the Blockified Single Product Template.
2. Create a product with password protection.
3. Visit the Product.
4. Ensure that the password form is added and the other blocks related to the Single Product Template are not visible.


## Screenshots or screencast



![image](https://github.com/woocommerce/woocommerce-blocks/assets/4463174/3242cb77-e61d-4422-84f6-3d2ce066e1c3)


## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Fix: Implement password protection on the blockified Single Product Template.
